edgelet/iotedge/src/check/checks/container_connect_upstream.rs

```diff
@@ -105,12 +105,17 @@ impl ContainerConnectUpstream {
             return Ok(CheckResult::Skipped);
         };
 
-        let upstream_hostname = if let Some(iothub_hostname) = &check.iothub_hostname {
+        let parent_hostname: String;
```

Comment:
is this needed, can it just return &upstream_hostname.to_string() ?

Reply:
I tried but I get: "creates a temporary which is freed while still in use"